packages/mongo/oplog_observe_driver.js

```diff
@@ -137,24 +137,37 @@ OplogObserveDriver = function (options) {
       var fence = DDPServer._CurrentWriteFence.get();
       if (!fence)
         return;
-      var write = fence.beginWrite();
-      // This write cannot complete until we've caught up to "this point" in the
-      // oplog, and then made it back to the steady state.
-      Meteor.defer(function () {
+
+      if (fence._oplogObserveDrivers) {
+        if (!_.contains(fence._oplogObserveDrivers, self))
+          fence._oplogObserveDrivers.push(self);
+
+        return;
+      }
+
+      fence._oplogObserveDrivers = [self];
+
+      fence.onArmed(function () {
+        // This fence cannot fire until we've caught up to "this point" in the
+        // oplog, and all observers made it back to the steady state.
         self._mongoHandle._oplogHandle.waitUntilCaughtUp();
-        if (self._stopped) {
-          // We're stopped, so just immediately commit.
-          write.committed();
-        } else if (self._phase === PHASE.STEADY) {
-          // Make sure that all of the callbacks have made it through the
-          // multiplexer and been delivered to ObserveHandles before committing
-          // writes.
-          self._multiplexer.onFlush(function () {
-            write.committed();
-          });
-        } else {
-          self._writesToCommitWhenWeReachSteady.push(write);
-        }
+
+        _.each(fence._oplogObserveDrivers, function (driver) {
```

Comment:
are we 100% confident that `fence._oplogObserveDrivers` will not be added to once the fence is armed?  I think so but want to make sure we've thought this through.